src/main/java/net/dv8tion/jda/core/entities/Game.java

```diff
@@ -15,40 +15,99 @@
  */
 package net.dv8tion.jda.core.entities;
 
-import net.dv8tion.jda.core.entities.impl.GameImpl;
 import net.dv8tion.jda.core.utils.Checks;
 
+import java.util.Objects;
+
 /**
  * Represents a Discord {@link net.dv8tion.jda.core.entities.Game Game}.
  * <br>This should contain all information provided from Discord about a Game.
  *
  * @since  2.1
  * @author John A. Grosh
  */
-public interface Game
+public class Game
 {
+    protected final String name;
+    protected final String url;
+    protected final Game.GameType type;
+
+    protected Game(String name)
+    {
+        this(name, null, GameType.DEFAULT);
+    }
+
+    protected Game(String name, String url)
+    {
+        this(name, url, GameType.TWITCH);
+    }
+
+    protected Game(String name, String url, GameType type)
+    {
+        this.name = name;
+        this.url = url;
+        this.type = type;
+    }
 
     /**
      * The displayed name of the {@link net.dv8tion.jda.core.entities.Game Game}. If no name has been set, this returns null.
      *
      * @return Possibly-null String containing the Game's name.
      */
-    String getName();
+    public String getName()
+    {
+        return name;
+    }
 
     /**
      * The URL of the {@link net.dv8tion.jda.core.entities.Game Game} if the game is actually a Stream.
      * <br>This will return null for regular games.
      *
      * @return Possibly-null String containing the Game's URL.
      */
-    String getUrl();
+    public String getUrl()
+    {
+        return url;
+    }
 
     /**
      * The type of {@link net.dv8tion.jda.core.entities.Game Game}.
      *
      * @return Never-null {@link net.dv8tion.jda.core.entities.Game.GameType GameType} representing the type of Game
      */
-    GameType getType();
+    public GameType getType()
+    {
+        return type;
+    }
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if (!(o instanceof Game))
+            return false;
+
+        Game oGame = (Game) o;
+        if (oGame.getType() != type)
```

Comment:
Never get used to how github posts snippets... (and not being able to set a range yourself)